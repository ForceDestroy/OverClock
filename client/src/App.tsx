import { BrowserRouter, Route, Routes, useLocation } from 'react-router-dom';
import { Box, Grommet } from 'grommet';
import './App.css';

import Home from './pages/Home';
import NotFound from './pages/NotFound';
import Login from './pages/Login';
import EmployeeDashboard from './pages/EmployeeDashboard';
import EmployerDashboard from './pages/EmployerDashboard';
import NavigationTopBar from './components/NavigationTopBar';
import ProfilePage from './pages/ProfilePage';
import Timesheets from './pages/Timesheets';
import Schedule from './pages/Schedule';
import TimeOff from './pages/TimeOff';
import EmployerTimesheets from './pages/EmployerTimesheets';
import Payslip from './pages/Payslip';
import EmployerRequests from './pages/EmployerRequests';
import CompanyPolicy from './pages/CompanyPolicy';

export function App() {
  const { pathname } = useLocation();
  const toggleNav = pathname === '/Login' || pathname === '/';
  const availableRoutes = [
    '/employeedashboard',
    '/employeraccount',
    '/employerdashboard',
    '/employer/timesheets',
    '/employer/requests',
    '/application',
    '/applicationreferral',
    '/applicationmanagement',
    '/companydashboard',
    '/employeedashboard/companypolicy',
    '/profilepage',
    '/employeedashboard/timesheets',
    '/employeedashboard/schedule',
    '/employeedashboard/payslip',
    '/employeedashboard/timeoff',
  ];
  const errorRoute = availableRoutes.includes(pathname.toLowerCase());
  return (
    <div className="App">
      <Grommet full>
        {!toggleNav && errorRoute && <NavigationTopBar />}
        <Box direction="row" height={{ min: '90vh' }} background="light-1">
          <Routes>
            <Route path="/" element={<Home />} />
            <Route path="/Login" element={<Login />} />
            <Route path="/EmployeeDashboard" element={<EmployeeDashboard />} />
            <Route path="/EmployerDashboard" element={<EmployerDashboard />} />
            <Route path="/Employer/Requests" element={<EmployerRequests />} />
            <Route
              path="/Employer/Timesheets"
              element={<EmployerTimesheets />}
            />
            <Route
              path="/EmployeeDashboard/CompanyPolicy"
              element={<CompanyPolicy />}
            />
            <Route path="/ProfilePage" element={<ProfilePage />} />
            <Route
              path="/EmployeeDashboard/Timesheets"
              element={<Timesheets />}
            />
            <Route path="/EmployeeDashboard/Schedule" element={<Schedule />} />
            <Route path="/EmployeeDashboard/Payslip" element={<Payslip />} />
            <Route path="/EmployeeDashboard/TimeOff" element={<TimeOff />} />
            <Route path="*" element={<NotFound />} />
          </Routes>
        </Box>
      </Grommet>
    </div>
  );
}

export function WrappedApp() {
  return (
    <BrowserRouter>
      <App />
    </BrowserRouter>
  );
}
