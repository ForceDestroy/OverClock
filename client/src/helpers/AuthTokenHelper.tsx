/* eslint-disable consistent-return */
export default class AuthTokenHelper {
  static getFromLocalStorage = (): string | null => {
    try {
      if (localStorage.getItem('JWT_OC') === null) {
        throw ReferenceError();
      }
      const token = localStorage.getItem('JWT_OC');

      return token;
    } catch (error: unknown) {
      return null;
    }
  };

  static saveToLocalStorage = (token: string, timestamp?: string) => {
    try {
      localStorage.setItem('JWT_OC', token);
      if (timestamp) {
        localStorage.setItem('JWT_OC_TIMESTAMP', timestamp);
      } else {
        localStorage.setItem(
          'JWT_OC_TIMESTAMP',
          new Date().getTime().toString()
        );
      }
    } catch (error: unknown) {
      return null;
    }
  };

  static updateInLocalStorage = (token: string, timestamp?: string) => {
    try {
      if (localStorage.getItem('JWT_OC') === null) {
        throw ReferenceError();
      }
      localStorage.removeItem('JWT_OC');
      localStorage.setItem('JWT_OC', token);
      localStorage.removeItem('JWT_OC_TIMESTAMP');
      if (timestamp) {
        localStorage.setItem('JWT_OC_TIMESTAMP', timestamp);
      } else {
        localStorage.setItem(
          'JWT_OC_TIMESTAMP',
          new Date().getTime().toString()
        );
      }
    } catch (error: unknown) {
      return null;
    }
  };

  static removeFromLocalStorage = () => {
    try {
      if (localStorage.getItem('JWT_OC') === null) {
        throw ReferenceError();
      }
      localStorage.removeItem('JWT_OC');
      localStorage.removeItem('JWT_OC_TIMESTAMP');
      localStorage.removeItem('JWT_OC_NAME');
      localStorage.removeItem('JWT_OC_ACCESS_LEVEL');
      localStorage.removeItem('JWT_OC_USER_ID');
    } catch (error: unknown) {
      return null;
    }
    return null;
  };

  static validateToken = (): boolean => {
    try {
      if (localStorage.getItem('JWT_OC') === null) {
        throw ReferenceError();
      }
      const tokenTimestamp = localStorage.getItem('JWT_OC_TIMESTAMP');
      const timestamp = Date.parse(tokenTimestamp || '');

      if (timestamp + 30 * 60 * 1000 < new Date().getTime()) {
        this.removeFromLocalStorage();
        return false;
      }

      return true;
    } catch (error: unknown) {
      return false;
    }
  };

  static setName = (name: string) => {
    try {
      if (localStorage.getItem('JWT_OC') === null) {
        throw ReferenceError();
      }
      localStorage.setItem('JWT_OC_NAME', name);
    } catch (error: unknown) {
      return null;
    }
  };

  static getName = (): string | null => {
    try {
      if (localStorage.getItem('JWT_OC') === null) {
        throw ReferenceError();
      }
      const name = localStorage.getItem('JWT_OC_NAME');

      return name;
    } catch (error: unknown) {
      return null;
    }
  };

  static setAccessLevel = (accessLevel: string) => {
    try {
      if (localStorage.getItem('JWT_OC') === null) {
        throw ReferenceError();
      }
      localStorage.setItem('JWT_OC_ACCESS_LEVEL', accessLevel);
    } catch (error: unknown) {
      return null;
    }
  };

  static getAccessLevel = (): string | null => {
    try {
      if (localStorage.getItem('JWT_OC') === null) {
        throw ReferenceError();
      }
      const accessLevel = localStorage.getItem('JWT_OC_ACCESS_LEVEL');

      return accessLevel;
    } catch (error: unknown) {
      return null;
    }
  };

  static setUserId = (userId: string) => {
    try {
      if (localStorage.getItem('JWT_OC') === null) {
        throw ReferenceError();
      }
      localStorage.setItem('JWT_OC_USER_ID', userId);
    } catch (error: unknown) {
      return null;
    }
  };

  static getUserId = (): string | null => {
    try {
      if (localStorage.getItem('JWT_OC') === null) {
        throw ReferenceError();
      }
      const userId = localStorage.getItem('JWT_OC_USER_ID');

      return userId;
    } catch (error: unknown) {
      return null;
    }
  };
}
