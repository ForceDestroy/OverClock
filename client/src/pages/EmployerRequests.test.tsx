import { render, screen, waitFor, act } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import axios from 'axios';
import { MemoryRouter } from 'react-router-dom';
import { describe, it, vi } from 'vitest';

import { App } from '../App';
import AuthTokenHelper from '../helpers/AuthTokenHelper';

vi.mock('axios');
vi.mock('../helpers/AuthTokenHelper');

describe('Employer Requests', () => {
  afterEach(() => {
    vi.clearAllMocks();
    vi.resetAllMocks();
  });
  it('Renders Employer Requests to match snapshot', async () => {
    // Mock
    (axios.request as jest.Mock).mockReturnValue({
      data: [
        {
          id: 9,
          fromId: 'WEB_000000',
          toId: 'MNG_000000',
          fromName: 'web test',
          toName: 'DrAiman Hanna',
          body: '30',
          date: '2023-01-30T00:00:00',
          startTime: '2023-01-30T00:00:00',
          endTime: '2023-01-31T00:00:00',
          type: 'Schedule',
          status: 'Approved',
        },
        {
          id: 10,
          fromId: 'WEB_000000',
          toId: 'MNG_000000',
          fromName: 'notweb test',
          toName: 'DrAiman Hanna',
          body: '',
          date: '2023-01-24T00:00:00',
          startTime: '2023-01-24T00:00:00',
          endTime: '2023-01-25T00:00:00',
          type: 'Vacation',
          status: 'Acknowledged',
        },
        {
          id: 27,
          fromId: 'AAA_000004',
          toId: 'MNG_000000',
          fromName: 'Danny',
          toName: 'DrAiman Hanna',
          body: 'I need my paystubs',
          date: '2023-01-19T00:00:00',
          startTime: '2023-01-19T00:00:00',
          endTime: '2023-01-19T00:00:00',
          type: 'Sick',
          status: 'Pending',
        },
      ],
      status: 200,
      statusText: 'OK',
    });
    (AuthTokenHelper.validateToken as jest.Mock).mockReturnValue(
      'VerySecretToken'
    );

    // Arrange);

    const snap = render(
      <MemoryRouter initialEntries={['/Employer/Requests']}>
        <App />
      </MemoryRouter>
    );

    // Assert
    expect(snap).toMatchSnapshot();
  });
  it('Renders a list of all submitted requests', async () => {
    // Mock
    (axios.request as jest.Mock).mockReturnValue({
      data: [
        {
          id: 9,
          fromId: 'WEB_000000',
          toId: 'MNG_000000',
          fromName: 'web test',
          toName: 'DrAiman Hanna',
          body: '30',
          date: '2023-01-30T00:00:00',
          startTime: '2023-01-30T00:00:00',
          endTime: '2023-01-31T00:00:00',
          type: 'Schedule',
          status: 'Approved',
        },
        {
          id: 10,
          fromId: 'WEB_000000',
          toId: 'MNG_000000',
          fromName: 'notweb test',
          toName: 'DrAiman Hanna',
          body: '',
          date: '2023-01-24T00:00:00',
          startTime: '2023-01-24T00:00:00',
          endTime: '2023-01-25T00:00:00',
          type: 'Vacation',
          status: 'Acknowledged',
        },
        {
          id: 27,
          fromId: 'AAA_000004',
          toId: 'MNG_000000',
          fromName: 'Danny',
          toName: 'DrAiman Hanna',
          body: 'I need my paystubs',
          date: '2023-01-19T00:00:00',
          startTime: '2023-01-19T00:00:00',
          endTime: '2023-01-19T00:00:00',
          type: 'Sick',
          status: 'Pending',
        },
      ],
      status: 200,
      statusText: 'OK',
    });
    (AuthTokenHelper.validateToken as jest.Mock).mockReturnValue(
      'VerySecretToken'
    );

    // Arrange
    const user = userEvent.setup();

    render(
      <MemoryRouter initialEntries={['/Employer/Requests']}>
        <App />
      </MemoryRouter>
    );

    // Act
    await user.click(screen.getByText('Refresh'));

    // Assert
    expect(screen.getByText('Danny')).toBeInTheDocument();
    expect(screen.getByText('web test')).toBeInTheDocument();
    expect(screen.getByText('notweb test')).toBeInTheDocument();
    expect(screen.getByText('2023-01-19')).toBeInTheDocument();
    expect(screen.getByText('2023-01-24')).toBeInTheDocument();
    expect(screen.getByText('2023-01-30')).toBeInTheDocument();
    expect(screen.getByText('Schedule')).toBeInTheDocument();
    expect(screen.getByText('Vacation')).toBeInTheDocument();
    expect(screen.getByText('Sick')).toBeInTheDocument();
    expect(screen.getByText('Acknowledged')).toBeInTheDocument();
    expect(screen.getByText('Approved')).toBeInTheDocument();
    expect(screen.getByText('Pending')).toBeInTheDocument();
  });
  it('Renders a full detailed list of a single request', async () => {
    // Mock
    (axios.request as jest.Mock).mockReturnValue({
      data: [
        {
          id: 9,
          fromId: 'WEB_000000',
          toId: 'MNG_000000',
          fromName: 'web test',
          toName: 'DrAiman Hanna',
          body: '30',
          date: '2023-01-30T00:00:00',
          startTime: '2023-01-30T00:00:00',
          endTime: '2023-01-31T00:00:00',
          type: 'Schedule',
          status: 'Approved',
        },
        {
          id: 10,
          fromId: 'WEB_000000',
          toId: 'MNG_000000',
          fromName: 'notweb test',
          toName: 'DrAiman Hanna',
          body: '',
          date: '2023-01-24T00:00:00',
          startTime: '2023-01-24T00:00:00',
          endTime: '2023-01-25T00:00:00',
          type: 'Vacation',
          status: 'Acknowledged',
        },
        {
          id: 27,
          fromId: 'AAA_000004',
          toId: 'MNG_000000',
          fromName: 'Danny',
          toName: 'DrAiman Hanna',
          body: 'I need my paystubs',
          date: '2023-01-19T00:00:00',
          startTime: '2023-01-19T00:00:00',
          endTime: '2023-01-19T00:00:00',
          type: 'Sick',
          status: 'Pending',
        },
      ],
      status: 200,
      statusText: 'OK',
    });
    (AuthTokenHelper.validateToken as jest.Mock).mockReturnValue(
      'VerySecretToken'
    );

    // Arrange
    const user = userEvent.setup();

    render(
      <MemoryRouter initialEntries={['/Employer/Requests']}>
        <App />
      </MemoryRouter>
    );

    // Act
    await user.click(screen.getByText('Refresh'));
    (axios.request as jest.Mock).mockReturnValue({
      data: [],
      status: 200,
      statusText: 'OK',
    });
    await user.click(screen.getByText('Danny'));

    // Assert
    expect(screen.getByText('Name: Danny')).toBeInTheDocument();
    expect(screen.getByText('ID: AAA_000004')).toBeInTheDocument();
    expect(screen.getByText('Reason: I need my paystubs')).toBeInTheDocument();
    expect(screen.getByText('Type: Sick')).toBeInTheDocument();
    expect(screen.getByText('Status: Pending')).toBeInTheDocument();
  });
  it('Renders back to dashboard when the user has the wrong access level', async () => {
    // Mock
    (axios.request as jest.Mock).mockReturnValue({
      data: {
        id: 'WEB_000000',
        name: 'web test',
        email: 'webtest@gmail.com',
        password: 'securepassword',
        address: '31 Jump Street',
        birthday: '0001-01-01T00:00:00',
        phoneNumber: 123,
        themeColor: 0,
        vacationDays: 5,
        sickDays: 7,
      },
      status: 200,
      statusText: 'OK',
    });
    (AuthTokenHelper.getAccessLevel as jest.Mock).mockReturnValue('0');
    (AuthTokenHelper.validateToken as jest.Mock).mockReturnValue(
      'VerySecretToken'
    );

    // Arrange

    render(
      <MemoryRouter initialEntries={['/Employer/Requests']}>
        <App />
      </MemoryRouter>
    );

    // Assert
    expect(screen.getByText('Mode Durgaa')).toBeInTheDocument();
  });
});
