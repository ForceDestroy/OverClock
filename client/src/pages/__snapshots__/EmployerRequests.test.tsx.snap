// Vitest Snapshot v1

exports[`Employer Requests > Renders Employer Requests to match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div
          class="StyledGrommet-sc-19lkkz7-0 jGVJhU"
        >
          <div>
            <div
              class="StyledGrid-sc-1wofa1l-0 bySYOJ"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 dDSiKk"
              >
                <div
                  class="StyledBox-sc-13pk1d4-0 ibTKOe"
                >
                  <a
                    aria-label="home"
                    class="StyledAnchor-sc-1rp7lwl-0 iUqWmS"
                    data-testid="home"
                    href="/"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 iNDmPr"
                    >
                      <span
                        class="StyledText-sc-1sadyjn-0 cMFmcc"
                      >
                         
                        Home
                         
                      </span>
                    </div>
                  </a>
                  <div
                    class="StyledBox__StyledBoxGap-sc-13pk1d4-1 eYHYac"
                  />
                  <a
                    aria-label="Calendar"
                    class="StyledAnchor-sc-1rp7lwl-0 iUqWmS"
                    data-testid="Calendar"
                    href="/EmployeeDashboard/Schedule"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 iNDmPr"
                    >
                      <span
                        class="StyledText-sc-1sadyjn-0 cMFmcc"
                      >
                         
                        Calendar
                         
                      </span>
                    </div>
                  </a>
                  <div
                    class="StyledBox__StyledBoxGap-sc-13pk1d4-1 eYHYac"
                  />
                  <a
                    aria-label="MyJob"
                    class="StyledAnchor-sc-1rp7lwl-0 iUqWmS"
                    data-testid="MyJob"
                    href="/"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 iNDmPr"
                    >
                      <span
                        class="StyledText-sc-1sadyjn-0 cMFmcc"
                      >
                         
                        My Job
                         
                      </span>
                    </div>
                  </a>
                </div>
                <div
                  class="StyledBox-sc-13pk1d4-0 bqfCTq"
                >
                  <div
                    class="StyledBox-sc-13pk1d4-0 iNDmPr"
                  >
                    <span
                      class="StyledText-sc-1sadyjn-0 cdHUSp"
                    >
                       
                       
                    </span>
                    <span
                      class="StyledText-sc-1sadyjn-0 hpCRgD"
                    >
                       Sales Associate 
                    </span>
                  </div>
                  <div
                    class="StyledBox__StyledBoxGap-sc-13pk1d4-1 lbllaT"
                  />
                  <a
                    class="StyledAnchor-sc-1rp7lwl-0 iUqWmS"
                    href="/ProfilePage"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 kobaLM"
                    >
                      <div
                        class="StyledBox-sc-13pk1d4-0 ebPCjN StyledAvatar-sc-1suyamb-1"
                      />
                    </div>
                  </a>
                  <div
                    class="StyledBox__StyledBoxGap-sc-13pk1d4-1 lbllaT"
                  />
                  <button
                    class="StyledButton-sc-323bzc-0 kYtlmw"
                    data-testid="logout"
                    type="button"
                  >
                    <svg
                      aria-label="Logout"
                      class="StyledIcon-sc-ofa7kd-0 cnewlt"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M13 9V2H1v20h12v-7m9-3H5m12-5 5 5-5 5"
                        fill="none"
                        stroke="#231F20"
                        stroke-width="2"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="StyledBox-sc-13pk1d4-0 eETalo"
          >
            <div
              class="StyledGrid-sc-1wofa1l-0 bOqVlp"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 izOtJh"
              >
                <span
                  class="StyledText-sc-1sadyjn-0 eHAjbp"
                >
                  Mode Durgaa - Approve Requests
                </span>
                <div
                  class="StyledBox__StyledBoxGap-sc-13pk1d4-1 ceHRnz"
                />
                <div
                  class="StyledBox-sc-13pk1d4-0 kNlEFC"
                >
                  <button
                    class="StyledButton-sc-323bzc-0 ivrARX"
                    data-testid="refreshButton"
                    type="button"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 jzOfgr"
                    >
                      <svg
                        aria-label="Refresh"
                        class="StyledIcon-sc-ofa7kd-0 kKhdEN"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M20 8c-1.403-2.96-4.463-5-8-5a9 9 0 1 0 0 18 9 9 0 0 0 9-9m0-9v6h-6"
                          fill="none"
                          stroke="#000"
                          stroke-width="2"
                        />
                      </svg>
                      <div
                        class="StyledBox__StyledBoxGap-sc-13pk1d4-1 lbllaT"
                      />
                      <span
                        class="StyledText-sc-1sadyjn-0 hpCRgD"
                      >
                        Refresh
                      </span>
                    </div>
                  </button>
                </div>
                <div
                  class="StyledBox__StyledBoxGap-sc-13pk1d4-1 ceHRnz"
                />
                <table
                  class="StyledTable-sc-1m3u5g-6 YPHTu StyledDataTable-sc-xrlyjm-0 hmFzTO"
                  data-testid="timesheetDataTable"
                >
                  <thead
                    class="StyledTable__StyledTableHeader-sc-1m3u5g-4 StyledDataTable__StyledDataTableHeader-sc-xrlyjm-4"
                  >
                    <tr
                      class="StyledTable__StyledTableRow-sc-1m3u5g-2 StyledDataTable__StyledDataTableRow-sc-xrlyjm-2 hSlrxd"
                    >
                      <th
                        class="StyledTable__StyledTableCell-sc-1m3u5g-0 bLGA-dP StyledDataTable__StyledDataTableCell-sc-xrlyjm-6 dsFDzi"
                        scope="col"
                      >
                        <div
                          class="StyledBox-sc-13pk1d4-0 fqUnkf"
                        >
                          <span
                            class="StyledText-sc-1sadyjn-0 hpCRgD"
                          >
                            Date
                          </span>
                        </div>
                      </th>
                      <th
                        class="StyledTable__StyledTableCell-sc-1m3u5g-0 bLGA-dP StyledDataTable__StyledDataTableCell-sc-xrlyjm-6 dsFDzi"
                        scope="col"
                      >
                        <div
                          class="StyledBox-sc-13pk1d4-0 fqUnkf"
                        >
                          <span
                            class="StyledText-sc-1sadyjn-0 hpCRgD"
                          >
                            Name
                          </span>
                        </div>
                      </th>
                      <th
                        class="StyledTable__StyledTableCell-sc-1m3u5g-0 bLGA-dP StyledDataTable__StyledDataTableCell-sc-xrlyjm-6 dsFDzi"
                        scope="col"
                      >
                        <div
                          class="StyledBox-sc-13pk1d4-0 fqUnkf"
                        >
                          <span
                            class="StyledText-sc-1sadyjn-0 hpCRgD"
                          >
                            Type
                          </span>
                        </div>
                      </th>
                      <th
                        class="StyledTable__StyledTableCell-sc-1m3u5g-0 bLGA-dP StyledDataTable__StyledDataTableCell-sc-xrlyjm-6 dsFDzi"
                        scope="col"
                      >
                        <div
                          class="StyledBox-sc-13pk1d4-0 fqUnkf"
                        >
                          <span
                            class="StyledText-sc-1sadyjn-0 hpCRgD"
                          >
                            Status
                          </span>
                        </div>
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    class="StyledTable__StyledTableBody-sc-1m3u5g-3 StyledDataTable__StyledDataTableBody-sc-xrlyjm-3 eaVduh"
                    tabindex="0"
                  />
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div
        class="StyledGrommet-sc-19lkkz7-0 jGVJhU"
      >
        <div>
          <div
            class="StyledGrid-sc-1wofa1l-0 bySYOJ"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 dDSiKk"
            >
              <div
                class="StyledBox-sc-13pk1d4-0 ibTKOe"
              >
                <a
                  aria-label="home"
                  class="StyledAnchor-sc-1rp7lwl-0 iUqWmS"
                  data-testid="home"
                  href="/"
                >
                  <div
                    class="StyledBox-sc-13pk1d4-0 iNDmPr"
                  >
                    <span
                      class="StyledText-sc-1sadyjn-0 cMFmcc"
                    >
                       
                      Home
                       
                    </span>
                  </div>
                </a>
                <div
                  class="StyledBox__StyledBoxGap-sc-13pk1d4-1 eYHYac"
                />
                <a
                  aria-label="Calendar"
                  class="StyledAnchor-sc-1rp7lwl-0 iUqWmS"
                  data-testid="Calendar"
                  href="/EmployeeDashboard/Schedule"
                >
                  <div
                    class="StyledBox-sc-13pk1d4-0 iNDmPr"
                  >
                    <span
                      class="StyledText-sc-1sadyjn-0 cMFmcc"
                    >
                       
                      Calendar
                       
                    </span>
                  </div>
                </a>
                <div
                  class="StyledBox__StyledBoxGap-sc-13pk1d4-1 eYHYac"
                />
                <a
                  aria-label="MyJob"
                  class="StyledAnchor-sc-1rp7lwl-0 iUqWmS"
                  data-testid="MyJob"
                  href="/"
                >
                  <div
                    class="StyledBox-sc-13pk1d4-0 iNDmPr"
                  >
                    <span
                      class="StyledText-sc-1sadyjn-0 cMFmcc"
                    >
                       
                      My Job
                       
                    </span>
                  </div>
                </a>
              </div>
              <div
                class="StyledBox-sc-13pk1d4-0 bqfCTq"
              >
                <div
                  class="StyledBox-sc-13pk1d4-0 iNDmPr"
                >
                  <span
                    class="StyledText-sc-1sadyjn-0 cdHUSp"
                  >
                     
                     
                  </span>
                  <span
                    class="StyledText-sc-1sadyjn-0 hpCRgD"
                  >
                     Sales Associate 
                  </span>
                </div>
                <div
                  class="StyledBox__StyledBoxGap-sc-13pk1d4-1 lbllaT"
                />
                <a
                  class="StyledAnchor-sc-1rp7lwl-0 iUqWmS"
                  href="/ProfilePage"
                >
                  <div
                    class="StyledBox-sc-13pk1d4-0 kobaLM"
                  >
                    <div
                      class="StyledBox-sc-13pk1d4-0 ebPCjN StyledAvatar-sc-1suyamb-1"
                    />
                  </div>
                </a>
                <div
                  class="StyledBox__StyledBoxGap-sc-13pk1d4-1 lbllaT"
                />
                <button
                  class="StyledButton-sc-323bzc-0 kYtlmw"
                  data-testid="logout"
                  type="button"
                >
                  <svg
                    aria-label="Logout"
                    class="StyledIcon-sc-ofa7kd-0 cnewlt"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M13 9V2H1v20h12v-7m9-3H5m12-5 5 5-5 5"
                      fill="none"
                      stroke="#231F20"
                      stroke-width="2"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 eETalo"
        >
          <div
            class="StyledGrid-sc-1wofa1l-0 bOqVlp"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 izOtJh"
            >
              <span
                class="StyledText-sc-1sadyjn-0 eHAjbp"
              >
                Mode Durgaa - Approve Requests
              </span>
              <div
                class="StyledBox__StyledBoxGap-sc-13pk1d4-1 ceHRnz"
              />
              <div
                class="StyledBox-sc-13pk1d4-0 kNlEFC"
              >
                <button
                  class="StyledButton-sc-323bzc-0 ivrARX"
                  data-testid="refreshButton"
                  type="button"
                >
                  <div
                    class="StyledBox-sc-13pk1d4-0 jzOfgr"
                  >
                    <svg
                      aria-label="Refresh"
                      class="StyledIcon-sc-ofa7kd-0 kKhdEN"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20 8c-1.403-2.96-4.463-5-8-5a9 9 0 1 0 0 18 9 9 0 0 0 9-9m0-9v6h-6"
                        fill="none"
                        stroke="#000"
                        stroke-width="2"
                      />
                    </svg>
                    <div
                      class="StyledBox__StyledBoxGap-sc-13pk1d4-1 lbllaT"
                    />
                    <span
                      class="StyledText-sc-1sadyjn-0 hpCRgD"
                    >
                      Refresh
                    </span>
                  </div>
                </button>
              </div>
              <div
                class="StyledBox__StyledBoxGap-sc-13pk1d4-1 ceHRnz"
              />
              <table
                class="StyledTable-sc-1m3u5g-6 YPHTu StyledDataTable-sc-xrlyjm-0 hmFzTO"
                data-testid="timesheetDataTable"
              >
                <thead
                  class="StyledTable__StyledTableHeader-sc-1m3u5g-4 StyledDataTable__StyledDataTableHeader-sc-xrlyjm-4"
                >
                  <tr
                    class="StyledTable__StyledTableRow-sc-1m3u5g-2 StyledDataTable__StyledDataTableRow-sc-xrlyjm-2 hSlrxd"
                  >
                    <th
                      class="StyledTable__StyledTableCell-sc-1m3u5g-0 bLGA-dP StyledDataTable__StyledDataTableCell-sc-xrlyjm-6 dsFDzi"
                      scope="col"
                    >
                      <div
                        class="StyledBox-sc-13pk1d4-0 fqUnkf"
                      >
                        <span
                          class="StyledText-sc-1sadyjn-0 hpCRgD"
                        >
                          Date
                        </span>
                      </div>
                    </th>
                    <th
                      class="StyledTable__StyledTableCell-sc-1m3u5g-0 bLGA-dP StyledDataTable__StyledDataTableCell-sc-xrlyjm-6 dsFDzi"
                      scope="col"
                    >
                      <div
                        class="StyledBox-sc-13pk1d4-0 fqUnkf"
                      >
                        <span
                          class="StyledText-sc-1sadyjn-0 hpCRgD"
                        >
                          Name
                        </span>
                      </div>
                    </th>
                    <th
                      class="StyledTable__StyledTableCell-sc-1m3u5g-0 bLGA-dP StyledDataTable__StyledDataTableCell-sc-xrlyjm-6 dsFDzi"
                      scope="col"
                    >
                      <div
                        class="StyledBox-sc-13pk1d4-0 fqUnkf"
                      >
                        <span
                          class="StyledText-sc-1sadyjn-0 hpCRgD"
                        >
                          Type
                        </span>
                      </div>
                    </th>
                    <th
                      class="StyledTable__StyledTableCell-sc-1m3u5g-0 bLGA-dP StyledDataTable__StyledDataTableCell-sc-xrlyjm-6 dsFDzi"
                      scope="col"
                    >
                      <div
                        class="StyledBox-sc-13pk1d4-0 fqUnkf"
                      >
                        <span
                          class="StyledText-sc-1sadyjn-0 hpCRgD"
                        >
                          Status
                        </span>
                      </div>
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="StyledTable__StyledTableBody-sc-1m3u5g-3 StyledDataTable__StyledDataTableBody-sc-xrlyjm-3 eaVduh"
                  tabindex="0"
                />
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
